index.js

```diff
@@ -0,0 +1,153 @@
+// See what all it can do here: https://dev.frontapp.com/plugin.html
+
+// This keeps back if Front has returned a conversation to the plugin.
+let hasConversation;
+
+// Listen for the `conversation` event from Front and print its contents, then load the contact to the plugin.
+Front.on('conversation', function (data) {
+  console.log('Event data', data);
+
+  // Set the state.
+  hasConversation = true;
+
+  // Load the Contact information based off of the event data. And set tab to Info.
+  loadContact(data.contact);
+  showInfo();
+});
+
+// Listen for the `no_conversation` event.  This can happen when opened to Inbox Zero.
+Front.on('no_conversation', function () {
+  console.log('No conversation');
+
+  // Set the state.
+  hasConversation = false;
+
+  // Display `No Contact` data and clear the notes and set the tab to Info.
+  displayContactInfo ("No Contact", "-");
+  displayCRMInfo("-", "-", "-");
+  clearNotes();
+  showInfo();
+});
+
+// Loads the contact once the body of the plugin is loaded.
+// This will call our mocked CRM service for data and then add the contact information and notes to the page.
+function loadContact(contact) {
+  // Display Front contact info.
+  displayContactInfo(contact.display_name, contact.handle);
+
+  // Build and display our CRM data.
+  const crmData = mockQueryCRM(contact.handle);
+  displayCRMInfo(crmData.info.id, crmData.info.location, crmData.info.status);
+
+  //  Load the notes from our CRM data.
+  displayNotes(crmData.notes);
+}
+
+// Create another note to add to the list.
+function createNote() {
+  if (!hasConversation) {
+    console.log('No conversation selected.');
+    return;
+  }
+
+  const note = mockPostNote();
+  displayNote(note);
+}
+
+// Displays Front contact information.
+function displayContactInfo (display_name, handle) {
+  const nameElement = document.getElementById("name");
+  const handleElement = document.getElementById("handle");
+
+  nameElement.innerHTML = display_name || handle;
+  handleElement.innerHTML = handle;
+}
+
+// Displays mocked CRM Info.
+function displayCRMInfo (id, location, status) {
+  const idElement = document.getElementById("id");
+  const locationElement = document.getElementById("location");
+  const statusElement = document.getElementById("status");
+
+  idElement.innerHTML = id;
+  locationElement.innerHTML = location;
+  statusElement.innerHTML = status;
+}
+
+// Displays the mocked CRM notes.
+function displayNotes(notes) {
+  // Reset the Notes column to make room for the newly found notes.
+  clearNotes();
+
+  // Add each Note to the Notes Column object.
+  notes.forEach(note => {
+    displayNote(note);
+  });
+}
+
+// Removes the currently displayed Notes.
+function clearNotes() {
+  const noteColumns = document.getElementById("notes");
+  noteColumns.innerHTML = "";
+}
+
+// Set the tab to Info and hide Notes.
```

Comment:
deferring to xav if this is the best way to do this (I know we have to do our best with just vanilla JS)

Reply:
Talked with Dennis about this, since this is so bare-bones, it sounds like it's the way to go for this.